hyperapp.d.ts

```diff
@@ -0,0 +1,128 @@
+export as namespace Hyperapp
+
+/** @namespace [Virtual DOM] */
+
+/** The Virtual DOM representation of an Element
+ *
+ * @memberOf [Virtual DOM]
+*/
+export interface VirtualNode<Data> {
+  tag: string
+  data: Data
+  children: VirtualNodeChild<{} | null>[]
+}
+
+/** In the Virtual DOM a Child could be either a Virtual Node or a string
+ *
+ * @memberOf [Virtual DOM]
+*/
+export type VirtualNodeChild<Data> = VirtualNode<Data> | string
+
+export interface Component<Data> {
+  /** A Component is a function that return a custom Virtual Node
+   *
+   * @memberOf [Virtual DOM]
+  */
+  (data: Data, children: VirtualNodeChild<{} | null>[]): VirtualNode<Data>
+}
+
+/** The soft way to create a Virtual Node
+ * @param tag       Either a tag name e.g. 'div'. Or a Component function
+ * @param data      Any valid HTML atributes, events, styles, and meta data
+ * @param children  The children of the VirtualNode
+ *
+ * @memberOf [Virtual DOM]
+*/
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?:
+    | VirtualNodeChild<{} | null>[]
+    | VirtualNodeChild<{} | null>
+    | number
+): VirtualNode<Data>
+
+/** @namespace [Application] */
+
+export interface State {}
+
+export interface Thunk {
+  (update: Function): any | null | void
+}
+
+export type ActionsResult<State extends Hyperapp.State> =
+  | Partial<State>
+  | Thunk
+  | {}
+  | null
+  | void
+
+
+export type ActionsCallers<
```

Comment:
@Swizz What if we call these VirtualActions? 

/cc @andyrj @pspeter3

Reply:
For naming I'll let you choose